src/codegen/optimizer.lisp

```diff
@@ -316,6 +320,106 @@
      (list
       (action (:after node-application) #'rewrite-application)))))
 
+(defun aggressive-heuristic (node)
+  (and (<= (count-applications node)
+           8)
+       (<= (count-nodes node)
+           32)))
+
+(defun heuristic-inline-applications (node
+                                      env
+                                      &key
+                                        (max-depth  16)
+                                        (max-unroll 4)
+                                        (heuristic  #'aggressive-heuristic))
```

Comment:
Now that the inliner inlines direct applications just the same as applications, it takes forever with the heuristic `(constantly t)`, even as low as `:max-depth 2 :max-unroll 1`. It seems that some of the tests must have significantly branching recursive functions with large bodies.